docs/contributing-to-analyses/analysis-modules/script-structure.md

```diff
@@ -0,0 +1,124 @@
+
+Below, we provide a few additional considerations that are specific to scripts in general, and then specific considerations for R and Python.
+
+### Considerations for any kind of script
+
+- The first line of your R script should be a ["shebang,"](https://linuxhandbook.com/shebang/) which stands for "ha**sh** (`#`) **bang** (`!`)".
+    - This makes the script executable by giving the exact path to the interpreter to use when running the script.
+    - Below, we provide specific instructions for how to write this line in [R scripts](#r-shebang) and [Python scripts](#python-shebang).
+- Your opening comments should be a brief, 1-3 sentences describing the purpose of the script.
+    - These comments _are complementary to_ but do not replace [additional documentation in your module's `README.md`](documenting-analysis.md).
+- If you need to import a separate script with reusable functions, you should import it in the section where you load other packages.
+- In most cases, your script should not hardcode input and output file paths.
+Instead, we recommend using an argument parser to read these paths in from the command line.
+    - Below, we provide recommendations for argument parsing in [R scripts](#r-arguments) and [Python scripts](#python-arguments).
+ - You may still want to export session information at the end of your script.
+See our additional documentation on how to do this for [R scripts](../determining-requirements/determining-software-requirements.md#in-a-script) and [Python scripts](../determining-requirements/determining-software-requirements.md#in-python-scripts).
+
+### Considerations for R scripts
+
+#### Shebang { #r-shebang }
+
+Your R scripts should always start with this line:
+
+```r
+#!/usr/bin/env Rscript
+```
+
+#### Parse input arguments { #r-arguments }
+
+We recommend using the [`optparse` package](https://cran.r-project.org/web/packages/optparse/index.html) to parse input arguments in R scripts.
+
+For example, after loading the `optparse` package, you can define input options with the [`make_option()`](https://rdrr.io/cran/optparse/man/add_make_option.html) function, and then use the [`parse_args()`](https://rdrr.io/cran/optparse/man/parse_args.html) function to parse them into a usable list.
+
+```r
+# Set up arguments
+option_list <- list(
+  make_option(
+    # Users can specify this argument with either `-i` or `--input_rds_file`
+    opt_str = c("-i", "--input_rds_file"),
+    # The provided value will be stored as a character variable called `input_anndata_file`
+    type = "character",
+    dest = "input_rds_file",
+    # Description of the argument
+    help = "path to input RDS file for analysis"
+  ),
+  ##  define additional options here ##
+)
+
+# Parse arguments into a usable list
+opt_list <- parse_args(OptionParser(option_list = option_list))
+```
+
+In the above example, you can access the parsed argument as `opt_list$input_rds_file`.
```

Comment:
I think being less verbose here is worth the cost, as `option_list` and `opt_list` are way too similar.
````suggestion
opts <- parse_args(OptionParser(option_list = option_list))
```

In the above example, you can access the parsed argument as `opts$input_rds_file`.
````